### PR TITLE
adds new markup for feature blocks and documentation

### PR DESCRIPTION
**Pull request by leejordan**
**Summary**

Refactors existing feature banners

Adds several new feature layout options:
- feature strips
- feature full width banner
- feature images

**Notable Changes**

existing feature banners have been refactored to avoid css spaghetti - this is a breaking change and will require new markup

**QA**

Check all new feature options in multiple browsers and devices. Make sure text is legible and everything is roughly consistent visually.

**Screenshots**

pistachio---layouts-feature-block-examples

### CODE REVIEW

File: less/modules/features.less [hunk 1]
```diff
@@ -120,17 +165,40 @@
...
+    @media (max-width: @screen-sm-max) {
+        width: 100% !important;
+        margin-top: 40%;
+        padding: @dimension-padding-lg !important;
+        background-color: @colour-grey-lightest !important;
+        background-image: none !important;
+        color: @text-colour !important;
+        text-shadow: none !important;
+        .reset-filter();
+    }
```
Comment (james-weaver) [inline less/modules/features.less]: `!important` 😢

File: site/docs/views/examples/features/banner-dark-right.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="feature-banner feature-banner--right" style="background-image:url('/images/banners/banner-yoga-right.jpg');">
+    <div class="feature-banner__content feature-banner__content--dark feature-banner__content--right">
+        <h2>good just got exciting</h2>
+        <p class="lead">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum.</p>
+        <a href="#0" class="button button--primary button--continue button--lg button--block-xs">get started</a>
+    </div>
+</div>
+
```
Comment (james-weaver) [inline site/docs/views/examples/features/banner-dark-right.html]: screen shot 2016-11-25 at 16 06 27

Text appears to start before the gradient.

File: site/docs/views/patterns_features.hbs [hunk 3]
```diff
@@ -5,12 +5,8 @@
...
 </ul>
 
-<h2 id="banner" class="cd-text cd-title">feature banner</h2>
-<p class="cd-text">Feature banners are designed to work with an image that is 1200px wide and 300px high. For art direction you should consider the right two thirds of the image as the focus as demonstrated below.</p>
-
-<img src="/images/banners/banner-template-simple.jpg" class="img-responsive">
-
-<p class="cd-text">Feature banners should always be full width and have several options for content so please visit the <a href="/layouts/feature-block-examples#banner">feature tile example page</a> for a demo.</p>
+<h2 id="banner" class="cd-text cd-title">feature banners</h2>
+<p class="cd-text">Feature banners have many options for content and fixed requirements for image sizes so please visit the dedicated <a href="/layouts/feature-block-examples">feature tile example page</a> for examples.</p>
```
Comment (james-weaver) [inline site/docs/views/patterns_features.hbs]: Url is `feature-block-examples` but the title is `feature-tile-examples`.

File: site/docs/views/layouts_feature-block-examples.hbs [hunk 4]
```diff
@@ -23,21 +25,125 @@
...
+                {{{ getFile 'site/docs/views/examples/features/banner-full-dark-right.html' }}}
+                {{#codeBlock}}
+                    {{{ getFile 'site/docs/views/examples/features/banner-full-dark-right.html' }}}
+                {{/codeBlock}}
+
+                <h3 class="cd-title">With a content panel aligned center</h3>
+                <h3 class="cd-title">image size: 1600px x 800px</h3>
+                <p class="cd-text">There is a unique image size for full width banners. For art direction you should be aware that the red section will always be visible on all viewport sizes while the entire image is visible at smaller viewport sizes.</p>
+                <img src="/images/banners/banner-template-full-center.jpg" class="img-responsive">
+                <br>
```
Comment (james-weaver) [inline site/docs/views/layouts_feature-block-examples.hbs]: screen shot 2016-11-25 at 16 24 46
Stretched in IE 11

File: less/modules/features.less (hunk 1 above)
Comment (leejordan) [inline less/modules/features.less]: I meant to tidy these up, will do so.

File: site/docs/views/patterns_features.hbs (hunk 3 above)
Comment (leejordan) [inline site/docs/views/patterns_features.hbs]: This page is linked to elsewhere and I didn't want to break that. I will add a 301 maybe. Maybe not.

File: site/docs/views/layouts_feature-block-examples.hbs (hunk 4 above)
Comment (leejordan) [inline site/docs/views/layouts_feature-block-examples.hbs]: annoying flexbox bug, I know how to fix this

File: less/modules/features.less (hunk 1 above)
Comment (leejordan) [inline less/modules/features.less]: these are gone

File: site/docs/views/examples/features/banner-dark-right.html (hunk 2 above)
Comment (leejordan) [inline site/docs/views/examples/features/banner-dark-right.html]: Fixed

File: site/docs/views/layouts_feature-block-examples.hbs (hunk 4 above)
Comment (leejordan) [inline site/docs/views/layouts_feature-block-examples.hbs]: fixed

File: site/docs/views/examples/features/banner-dark-right.html (hunk 2 above)
Comment (james-weaver) [inline site/docs/views/examples/features/banner-dark-right.html]: screen shot 2016-11-28 at 13 31 00

Comment (james-weaver) [inline site/docs/views/examples/features/banner-dark-right.html]: There still seems to be a problem with this banner.

Comment (leejordan) [inline site/docs/views/examples/features/banner-dark-right.html]: Possibly you didn't rebuild the css?

screen shot 2016-11-28 at 14 25 05

**Pull request merged**